yoyodyne/models/pointer_generator.py

```diff
@@ -427,16 +433,28 @@ def forward(
                 )
             else:
                 h_features, c_features = self.init_hiddens(len(batch))
-            predictions = self.decode(
-                source_encoded,
-                batch.source.mask,
-                batch.source.padded,
-                last_hiddens,
-                self.teacher_forcing if self.training else False,
-                features_enc=features_encoded,
-                features_mask=batch.features.mask,
-                target=batch.target.padded if batch.target else None,
-            )
+            if self.beam_width is not None and self.beam_width > 1:
+                # predictions, scores = self.beam_decode(
```

Comment:
Could we just let them call it and get the `NotImplementedError` from the base model? It seems harmless to do so rather than repeat ourselves on lines 444-446.

Ditto for the transducer and transformer.

Reply:
The transformer and transducer call the `beam_decode` function of the base model. However, the pointer generator LSTM calls the LSTM `beam_decode`, resulting in a failure. Therefore, I had to comment on the call and put there the NotImplementedError.

Reply:
Oh, that makes then that pointer-generator has to override it. Thanks for clarification, resolving.